openc3-cosmos-init/plugins/packages/openc3-cosmos-ace-diff/src/diff-match-patch.js

```diff
@@ -1020,6 +1022,13 @@ diff_match_patch.prototype.diff_cleanupSemanticLossless = function (diffs) {
       // First, shift the edit as far left as possible.
       var commonOffset = this.diff_commonSuffix(equality1, edit)
       if (commonOffset) {
+        // JMT: If the end of the common suffix is a newline then
+        // keep the newline because we want to favor trailing newlines
+        // to support the common use case of adding a line. Otherwise the
+        // algorithm shifts left to include the leading newline.
+        if (equality1[equality1.length - commonOffset] === '\n') {
+          commonOffset -= 1
+        }
```

Comment:
This is what was breaking the common diff use case of inserting a line. A diff of `added code\n` became `\nadded code`